src-manual/Vulkan/Requirements.hs

```diff
@@ -94,7 +126,9 @@ data Requirement where
   RequireInstanceSetting
     :: forall struct
     .  KnownInstanceSettingStruct struct
-    => { setInstanceSetting :: struct -> struct }
+    => { settingName        :: ByteString
+       , setInstanceSetting :: struct -> struct
+       }
```

Comment:
> I think it would be really good to replicate what the queue assignment
> stuff does. It takes the requirements packed into any `Traversable` and returns
> to the user the same shape but with proof that the queues have been assigned
> correctly at each index (in the form of a program to get that queue from the
> device, and the device params).
> 
> The way that would work here would be along the lines of:
> 
> ```haskell
> createInstanceFromRequirements
>   :: Traversable f
>   => f Requirements -- optional requirements
>   -> InstanceCreateInfo es
>   -> io (f RequirementResult, Maybe Instance)
> 

Ok, but the point was that the type `es` is created by inspecting the `Requirements`, so the user doesn't have to come up with the structure chain. I was hoping to entirely avoid having the user need to come up with structure chains if possible.

Reply:
I changed it so you directly specify the struct instead of providing a modification function, and I added a comment. I think that should be enough.